castervoice/apps/ide/ide.py

```diff
@@ -0,0 +1,94 @@
+#
+# __author__ = "lexxish"
+#
+from castervoice.apps.shared.directions import FORWARD, RIGHT, BACK, LEFT, UP, DOWN
+
+method = "(meth|method)"
+
+# general
+EXPAND_SELECTION = "expand [selection] [<n>]"
+SHRINK_SELECTION = "shrink [selection] [<n>]"
+SMART_AUTO_COMPLETE = "(skraken|smart kraken)"
+GENERATE_CODE = "jen code"
+QUICK_FIX = "quickfix"
+FORMAT_ALL_CODE = "format [code]"
+BUILD_PROJECT = "build"
+RUN_PROJECT = "run"
+DEBUG_PROJECT = "debug"
+BUILD_AND_RUN_PROJECT = "build and run"
+DEBUG_CURRENT_FILE = "debug file"
+RUN_CURRENT_FILE = "run file"
+NEXT_ERROR = "(%s error|error %s)" % (FORWARD, RIGHT)
+PREVIOUS_ERROR = "(%s error|error %s)" % (BACK, LEFT)
+REDO = "redo [<n>]"
+SHOW_SETTINGS = "[show] settings"
+RENAME_CURRENT_FILE = "file rename | rename file"
+
+# window navigation
+NEXT_TAB = "%s tab [<n>]|tab %s [<n>]" % (FORWARD, RIGHT)
+PREVIOUS_TAB = "%s tab [<n>]|tab %s [<n>]" % (BACK, LEFT)
```

Comment:
There are plenty of commands that should be common between all apps like next tab and previous tab. Should these get pulled out of both into a shared_commands file that browser_shared_commands etc pull from?

Reply:
yes I would think so